colcon_core/shell/template/prefix_util.py.em

```diff
@@ -256,22 +256,25 @@ def handle_dsv_types_except_source(type_, remainder, prefix):
         DSV_TYPE_PREPEND_NON_DUPLICATE,
         DSV_TYPE_PREPEND_NON_DUPLICATE_IF_EXISTS
     ):
-        env_name, value = remainder.split(';', 1)
-        if not value:
-            value = prefix
-        elif not os.path.isabs(value):
-            value = os.path.join(prefix, value)
-        if (
-            type_ == DSV_TYPE_PREPEND_NON_DUPLICATE_IF_EXISTS and
-            not os.path.exists(value)
-        ):
-            comment = 'skip extending {env_name} with not existing path: ' \
-                '{value}'.format_map(locals())
-            if _include_comments():
-                commands.append(
-                    FORMAT_STR_COMMENT_LINE.format_map({'comment': comment}))
-        else:
-            commands += _prepend_unique_value(env_name, value)
+        env_name_and_values = remainder.split(';')
+        env_name = env_name_and_values[0]
+        values = env_name_and_values[1:]
+        if 0 == len(values):
+            values.append(prefix)
```

Comment:
Since the `.dsv` file always contains a trailing semicolon if there are no actaul path values the `values` variable is never empty but always at least contains one empty string. So this logic doesn't make sense.

Reply:
Guarded against empty strings in the list d6fdbd5

Reply:
This is still wrong. The list can contain an empty value or multiple values where one could be empty. They must not be removed since they represent the prefix path.

Reply:
Oh, I thought that we'd only append the prefix path in the case there are no values provided.
I'll change the logic so every empty value is set to the prefix path.

Where is is documented that an empty value represents the prefix path (or that the prefix path is appended)? I think it might need to be updated along with this change.

Reply:
All values are simply either relative paths or absolute path. And relative paths are prefixed with the current prefix path to make them absolute.

Reply:
Might be nice to document the .dsv feature altogether somewhere, so it's easier for people to leverage it if as they need. But this is outside the scope of this PR.

I've updated again, using the prefix path in place of all empty values (c8b8dac).